File: typescript/reviewability_patterns.ts
/**
 * Benchmark patterns for TypeScript reviewability (future-maintainer change risk).
 *
 * Each block contains the EXACT OriginalCode from templates.go.
 */

// Mock service for pattern matching
declare const invoiceService: { execute(command: InvoiceCommand): unknown };

// ts-reviewability-parallel-state-medium - exact multi-line
type AccountStatus = "active" | "disabled";

interface Account {
  id: string;
  status: AccountStatus;
}

function isActive(account: Account): boolean {
  return account.status === "active";
}

// ts-reviewability-opaque-workflow-hard - exact multi-line
type InvoiceAction = "send" | "void";

interface InvoiceCommand {
  action: InvoiceAction;
  invoiceId: string;
  actorId: string;
}

function executeInvoiceCommand(command: InvoiceCommand) {
  return invoiceService.execute(command);
}

File: java/reviewability_patterns.java
// Benchmark patterns for Java reviewability (future-maintainer change risk).
// Each block contains the EXACT OriginalCode from templates.go.
package benchmarks;

import java.time.Duration;

class ReviewabilityPatterns {
    enum AccountStatus { ACTIVE, DISABLED }

    AccountStatus status;

    static class RetryPolicy {
        static RetryPolicy exponentialBackoff(Duration base, int maxAttempts) {
            return new RetryPolicy();
        }

        RetryPolicy() {}

        RetryPolicy(int maxAttempts, Duration window) {}
    }

    // java-reviewability-hidden-invariant-medium - exact
    RetryPolicy policy = RetryPolicy.exponentialBackoff(Duration.ofSeconds(2), 4);

    // java-reviewability-parallel-cache-hard - exact multi-line
    boolean isActive() {
        return status == AccountStatus.ACTIVE;
    }

    // java-fp-reviewability-typed-policy - exact multi-line
    void typedPolicy() {
        Duration safeBaseDelay = Duration.ofSeconds(2);
        int safeMaxAttempts = 4;
        RetryPolicy safePolicy = RetryPolicy.exponentialBackoff(safeBaseDelay, safeMaxAttempts);
    }
}
